csrc/fusion.cpp

```diff
@@ -571,7 +573,8 @@ void Fusion::registerExpr(Expr* expr) {
   }
 
   if (has_tv) {
-    resetTvUses();
+    // trigger a reset next time uses is needed
+    all_tv_uses_valid_ = false;
```

Comment:
If we can defer updating tv uses here, can we also do that for the other `resetTvUses()` in `Fusion::replaceOutput?`

Reply:
It looks to me like that would be fine. Good for completeness as that's the only other use outside of `Val::uses()`.